proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -925,9 +931,29 @@ def initialize(blk, initialize_guess=None, state_args=None, outlvl=idaeslog.NOTS
             solver : solver object or string indicating which solver to use during
                      initialization, if None provided the default solver will be used
                      (default = None)
+            fail_on_warning : boolean argument to fail or only produce  warning upon unsuccessful solve (default=False)
+
         Returns:
             None
         """
+        def _check_solve(results, checkpoint=None):
+            def _raise_if_infeasible(fail_flag):
+                msg = f"{checkpoint} failed. The solver failed to converge to an optimal solution. " \
+                      f"This suggests that the user provided infeasible inputs or that the model is poorly scaled."
+                if fail_flag is True:
+                    raise RuntimeError(msg)
+                elif fail_flag is False:
+                    init_log.warning(msg)
+                else:
+                    raise Exception(f'The fail_on_warning argument in the initialize method was set to {fail_flag}. '
+                                    f'fail_on_warning is a boolean argument. Set fail_on_warning to True or False.')
+            if checkpoint is None:
+                checkpoint = 'Initialization step'
+            if check_optimal_termination(results):
```

Comment:
What exactly does this method test for, and what conditions pass/fail? Notably, how does it handle IPOPTs "Solved to acceptable level" result?

My guess is that it is the same as what we would test, but we need to make sure.

Reply:
> This function returns True if the termination condition for the solver is 'optimal', 'locallyOptimal', or 'globallyOptimal', and the status is 'ok"

The source is [here](https://github.com/Pyomo/pyomo/blob/4f6f523483ffbf0eade463b1872885a140b607f1/pyomo/opt/results/solver.py).

Reply:
So, my only comment here is that we might be willing to accept an "Acceptable level" result from IPOPT, but I am not sure whether that would pass this test or not. However, that is also an edge case that we don't need to deal with immediately.